sgl-kernel/csrc/elementwise/activation.cu

```diff
@@ -13,33 +13,87 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <torch/all.h>
+
+#ifndef USE_ROCM
+
 #include <flashinfer/activation.cuh>
 
-#include "pytorch_extension_utils.h"
+#include "utils.h"
 
-using namespace flashinfer;
+#else
+#include "act_and_mul_internal.cuh"
+#endif
 
-__device__ __forceinline__ float silu(const float& val) {
-  return val / (1.0f + __expf(-val));
+// [flashinfer
+// activation(https://github.com/flashinfer-ai/flashinfer/blob/4e8eb1879f9c3ba6d75511e5893183bf8f289a62/csrc/activation.cu#L44)
+
+template <typename T>
+__device__ __forceinline__ T silu(const T& x) {
+#if USE_ROCM
+  float f32_val = castToFloat(x);
+  return castFromFloat<T>(f32_val / (1.0f + expf(-f32_val)));
```

Comment:
nit: Consider encapsulating both castFromFloat and static_cast into same-named functions earlier in the code.

Reply:
Thanks for the suggestion. I have refactored the script.